python/src/utils.h

```diff
@@ -15,6 +15,8 @@ namespace py = pybind11;
 using namespace mlx::core;
 
 using IntOrVec = std::variant<std::monostate, int, std::vector<int>>;
+using IntOrIntVec =
```

Comment:
I would probably write this inline until we have another version that uses the same variant. See for instance `pad` which has a unique variant as well. Next op that needs an `IntOrVecIntVec` could maybe add a name here and change both.